next/src/pages/boards/components/AllPosts.tsx

```diff
@@ -0,0 +1,68 @@
+import styles from "./AllPost.module.css";
+import Input from "../../../components/Input";
+import BasePost from "./BasePost";
+import { useEffect, useState } from "react";
+import { getPostsData } from "../../../api/api";
+import ToggleBtn from "../../../components/ToggleBtn";
+import Link from "next/link";
+import { PostData } from "../../../types";
+
+export default function AllPosts() {
+  const [postList, setPostList] = useState<PostData[]>([]);
+  const [orderBy, setOrderBy] = useState<"recent" | "like">("recent");
+  const [filterPostList, setFilterPostList] = useState<PostData[]>([]);
+  const [searchValue, setSearchValue] = useState<string>("");
+
+  const getPostListData = async (orderBy: "recent" | "like") => {
+    try {
+      const data: PostData[] = await getPostsData({
+        orderBy,
+        pageSize: 4,
+      });
+      setPostList(data);
+      setFilterPostList(data);
+    } catch (e) {
+      console.error("베이스 포스트 데이터를 패칭하는데 실패했습니다.", e);
+    }
+  };
+
+  const onChangeOrderBy = (orderByValue: "recent" | "like") => {
+    setOrderBy(orderByValue);
+  };
+
+  const handleSearch = () => {
+    const filtered = postList.filter((post) =>
+      post.content!.includes(searchValue)
+    );
+    setFilterPostList(filtered);
+  };
+
+  useEffect(() => {
+    getPostListData(orderBy);
+  }, [orderBy]);
+
+  return (
+    <div className={styles.container}>
+      <div className={styles.title}>
+        <h3>게시글</h3>
+        <Link href={"/"} className={styles.link}>
+          글쓰기
+        </Link>
+      </div>
+      <div className={styles.input}>
+        <Input
+          value={searchValue}
+          placeholder={"검색할 상품의 내용을 입력해주세요"}
```

Comment:
여기도 마찬가지로 string literal 그대로 활용해주세요
```suggestion
          placeholder="검색할 상품의 내용을 입력해주세요"
```